aiida_quantumespresso/parsers/parse_raw/ph.py

```diff
@@ -191,29 +145,48 @@ def parse_xml_matrices(tagname,target_tags):
     list_tuple = list(zip(*[iter(flat_array)]*3))
     return [list(i) for i in list_tuple]
 
-def parse_ph_text_output(lines):
-    """
-    Parses the stdout of QE-PH.
+def parse_ph_text_output(lines, logs):
+    """Parses the stdout of Quantum ESPRESSO ph.x.
 
     :param lines: list of strings, the file as read by readlines()
-
-    :return parsed_data: dictionary with parsed values.
-    :return critical_messages: a list with critical messages. If any is found in
-                               parsed_data['warnings'], the calculation is FAILED!
+    :return: dictionary with parsed values
     """
-    from aiida_quantumespresso.parsers.parse_raw.pw import parse_QE_errors
+    def detect_important_message(logs, line):
+
+        message_map = {
+            'error': {
+                'Maximum CPU time exceeded': 'ERROR_OUT_OF_WALLTIME',
+                'No convergence has been achieved': 'ERROR_CONVERGENCE_NOT_REACHED',
+            },
```

Comment:
No `'%%%%%%%%%%%%%%'`?

Reply:
I excluded this on purpose, just like for `PwParser`. I prefer to only add specific error messages, because only then can we do something useful with it.

Reply:
Makes sense. But don't we risk ignoring possibly important messages? Don't you want to log them as warnings? Or are they too much of a pain to parse?

Reply:
They are more difficult to parse, because there could be anything in between. That is of course "solvable" by getting everything between the markers. The main argument I find though is that these are errors that are critical, i.e. they will stop the calculation immediately. They are not warnings as such.

Reply:
Ok, so you're relying on 'JOB DONE' not being present in this case, which triggers 'ERROR_OUTPUT_STDOUT_INCOMPLETE'. Makes sense. Thanks.